desktop-widgets/subsurfacewebservices.cpp

```diff
@@ -91,15 +91,6 @@ static bool merge_locations_into_dives(void)
 						    time_during_dive_with_offset(dive, nextgpsfix->when, SAME_GROUP)) {
 							if (verbose)
 								qDebug() << "look at the next gps fix @" << get_dive_date_string(nextgpsfix->when);
-							/* first let's test if this one is during the dive */
```

Comment:
I struggle with this because this simply decides to optimize for one case over the other.
Yes, if the fix before the dive is within maybe 5 minutes I agree. But generally I'd say that the likelihood of the boat being in the right spot during the dive is overall higher than the likelihood of the boat being in the right spot before the dive - most cases your boat heads to the dive spot and then you get in. So ten minutes before the dive you could be quite a ways away.
So I'm not convinced that this is the right choice.

Reply:
> this simply decides to optimize for one case over the other.

Hmmm.  Not the idea. It was making the logic more general by choosing the gps fix closest to the entry point.

> So ten minutes before the dive you could be quite a ways away.

That's right, but that fix would be preferred **only** if the next fix is more than 10 minutes into the dive time, which means more than 20 minutes between fixes.

In the end, I think it's mostly a matter of a well configured background service. I've been using current logic set to 50m / 5minutes  without a flaw, until past summer while drift diving (and even then, deviation wasn't that big, at least not for trying to change the code).

Reply:
That is actually a good point. If we pick 5 minutes as the default frequency, then always picking the closest **is** the better choice.
And with that the second point below should be almost academic because you SHOULD have fixes during the dive which we'd prefer.